parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -192,5 +203,6 @@ public void close() throws IOException {
     flushRowGroup(true);
     writeStore.close();
     writer.end(metadata);
+    this.closed = true;
```

Comment:
If this is going to track whether the writer is closed, then it should make the close idempotent by checking the flag:

```java
  if (!closed) {
    this.closed = true;
    flushRowGroup(true);
    writeStore.close();
    writer.end(metadata);
  }
```

I would also add a precondition in `startRowGroup` that checks that `closed` is false.

Reply:
I'd also add :

```java
else {
  throw new IllegalStateException("writer is closed");
}
```

but note sure if it aligns with Iceberg's coding style